jupyter-extension/src/plugins/mount/mountDrive.ts

```diff
@@ -215,54 +210,89 @@ export class MountDrive implements Contents.IDrive {
     }
     this._previousFilter = newFilter;
 
-    this.setupScrollingHandler();
-
-    const {start} = this.getContentsStart();
-    const {end} = this.getContentsEnd();
     return {
       ...this._cache.shallowResponse,
-      content: this._cache.filteredContents.slice(start, end),
+      content: this._cache.filteredContents.slice(
+        this._index,
+        this._index + VISIBLE_CONTENT_LI_COUNT,
+      ),
     };
   }
 
-  private _fetchNextPage(
-    previousResponse: Contents.IModel,
+  private async _fetchNextPage(
```

Comment:
This was refactored primarily so that I could avoid the initial `_get` call and just use this function to start the chain of requests. It is functionally the same except the new snippet

```
    if (previousResponse) {
      await this.updateScrollPosition();
    }
```